debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSnapshotChangeEventSource.java

```diff
@@ -181,21 +179,24 @@ protected SnapshottingTask getSnapshottingTask(MongoDbPartition partition, Mongo
         final List<ReplicaSet> replicaSetSnapshots = new ArrayList<>();
         final MongoDbOffsetContext offsetContext = (MongoDbOffsetContext) previousOffset;
         try {
-            replicaSets.onEachReplicaSet(replicaSet -> {
-                MongoPreferredNode mongo = null;
+            for (var replicaSet : replicaSets.all()) {
+                RetryingMongoClient mongo = null;
                 try {
                     mongo = establishConnection(partition, replicaSet, ReadPreference.primaryPreferred());
                     final ReplicaSetOffsetContext rsOffsetContext = offsetContext.getReplicaSetOffsetContext(replicaSet);
                     if (mongo != null && isSnapshotExpected(mongo, rsOffsetContext)) {
                         replicaSetSnapshots.add(replicaSet);
                     }
                 }
+                catch (InterruptedException e) {
+                    // TODO: what to do here?
```

Comment:
@jpechane Here I am unsure what would be the correct reaction here. The method doesn't allow exceptions to be thrown and original behaviour ignored this exception (inside the client).